api/src/pcapi/scripts/remove_vo_vf_allocine_offers.py

```diff
@@ -0,0 +1,50 @@
+import time
+
+import sqlalchemy as sa
+
+from pcapi.core.offers.models import Offer
+from pcapi.core.providers.models import AllocinePivot
+from pcapi.core.search.backends import algolia
+from pcapi.core.search.backends import base
+from pcapi.models import db
+
+
+def remove_vo_vf_allocine_offers(dry_run: bool = True, batch_size: int = 10_000) -> None:
+    search_backend = base.SearchBackend() if dry_run else algolia.AlgoliaBackend()
+    query = (
+        AllocinePivot.query.filter(Offer.venueId == AllocinePivot.venueId)
+        .filter(Offer.isActive.is_(True))
+        .filter(sa.or_(Offer.name.endswith(" - VO"), Offer.name.endswith(" - VF")))
+        .with_entities(Offer.id)
+    )
+    start = time.time()
+    result = query.all()
+    after = time.time()
+    print(f"Initial query time: {after - start:.2f}s")
+    print(f"Found {len(result)} offers to update")
+
+    ids = [row[0] for row in result]
+    start_index = 0
+    while start_index < len(result):
```

Comment:
La query ci-dessus est évaluée ici mais elle prend plusieurs minutes à s’exécuter sur staging, du coup c’est plutôt celle là qu’il faudrait executer par batch non ? Par ailleurs je suis pas sur de comprendre l’utilité de la query ci-dessous, `unindex_offer_ids` n’attend que les ids des offres, mais du coup on les a déjà non ?

Reply:
À la louche quelque chose comme ça, pour tirer parti des index sur les id:

```suggestion
            query = (
                AllocinePivot.query.filter(Offer.venueId == AllocinePivot.venueId)
                .filter(Offer.isActive.is_(True))
                .filter(sa.or_(Offer.name.endswith(" - VO"), Offer.name.endswith(" - VF")))
                .filter(sa.and_(Offer.id > start_index, Offer.id < start_index + batch_size
                .with_entities(Offer.id, Offer.isActive)
            )
```

Par paquet de 10000 ça prend juste quelques secondes à s’executer.

Après si c’est juste un script oneshot, j’imagine qu’on se préoccupe moins du temps d’execution

Reply:
Ta proposition est mieux yes, en effet difficile d’arrêter la boucle de la mienne sans parcourir toutes les offres. Je me demandais surtout si la requête initiale ne risquait pas tout simplement de timeout